### PR TITLE
Decompose AdminTokenScopes and TopologyPanel into focused sub-components (#80)

### PR DESCRIPTION
**Pull request by dpage**
## Summary

- Decompose `AdminTokenScopes.tsx` (1,287 → 554 lines) into a `tokens/` directory with 8 focused sub-components: `tokenTypes`, `ScopeMultiSelect`, `ConnectionScopeTable`, `CreateTokenDialog`, `EditTokenDialog`, `CreatedTokenDialog`, `TokensTable`, `ApiUsageExample`.
- Decompose `TopologyPanel.tsx` (1,082 → 521 lines) into a `topology/` directory with 4 sub-components: `topologyHelpers`, `ServerManagementSection`, `RelationshipSection`, `RemoveServerDialog`.
- Add 216 new tests (126 for tokens, 90 for topology) — TopologyPanel had zero tests before this change.
- All new and modified files meet the 90% line coverage floor.

## Test plan

- [ ] Verify all 216 new sub-component tests pass (tokens + topology directories).
- [ ] Verify existing AdminTokenScopes tests (6 tests) still pass unchanged.
- [ ] Verify TopologyPanel.tsx at 90.59% coverage, topology/ at 99.21% coverage.
- [ ] Verify tokens/ sub-components at 90%+ coverage each.
- [ ] Verify ESLint passes with zero warnings on all changed files.
- [ ] Verify CI builds pass for Node.js 20 and 22.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Redesigned token management UI: create/edit/created dialogs, tokens table, copyable token dialog, connection-scoped access controls, API usage examples, reusable scope multi-select, and connection-scope table; new exported types and centralized token exports.
  * New topology panels: server management, relationships, and remove-server dialog with improved add/remove flows.

* **Refactor**
  * Topology panel reorganized; helper utilities extracted and re-exported.

* **Tests**
  * Extensive unit and UI tests covering tokens, topology, and helpers.

### CODE REVIEW

File: client/src/components/AdminPanel/tokens/CreateTokenDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+            onScopedConnectionsChange([
+                ...scopedConnections,
+                {
+                    id: connection.id,
+                    name: connection.name,
+                    access_level: maxLevel,
+                },
+            ]);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/AdminPanel/tokens/CreateTokenDialog.tsx:149]: This is pre-existing behavior from the original AdminTokenScopes.tsx (line 335 in the original: the catch block sets ownerConnections but not ownerConnectionLevels). This refactoring PR preserves the existing behavior as-is. The observation is valid and worth addressing in a separate issue, but it is out of scope for this decomposition-only change.

File: client/src/components/AdminPanel/tokens/TokensTable.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+                                                            adminPermissions={
+                                                                token.scope?.admin_permissions?.includes(
+                                                                    '*'
+                                                                )
+                                                                    ? [
+                                                                          'All Admin Permissions',
+                                                                      ]
+                                                                    : token.scope
+                                                                          ?.admin_permissions
+                                                            }
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/AdminPanel/tokens/TokensTable.tsx:262]: This is pre-existing behavior. The original AdminTokenScopes.tsx (line 714) already pre-labels wildcards as 'All MCP Privileges' before passing to EffectivePermissionsPanel. This decomposition preserves the original logic exactly. Worth a follow-up cleanup but not in scope here.

File: client/src/components/AdminPanel/tokens/TokensTable.tsx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                                                            isSuperuser={true}
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/AdminPanel/tokens/TokensTable.tsx:263]: This is pre-existing behavior. The original AdminTokenScopes.tsx (line 722) has the same isSuperuser={true} hardcoding. The refactoring preserves the original logic as-is. Worth a separate fix but not in scope for this decomposition PR.

File: client/src/components/topology/RelationshipSection.tsx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                                !selectedRelType
+                            }
+                            sx={{
+                                textTransform: 'none',
+                                whiteSpace: 'nowrap',
+                                height: 40,
+                            }}
+                        >
+                            Add
+                        </Button>
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/topology/RelationshipSection.tsx:276]: Fixed in 4911942 — added aria-label="Add relationship" to distinguish from the server Add button.

File: client/src/components/topology/RemoveServerDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                    disabled={removing}
+                >
+                    {removing ? (
+                        <CircularProgress
+                            size={18}
+                            sx={{ color: 'inherit' }}
+                        />
+                    ) : (
+                        'Remove'
+                    )}
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/topology/RemoveServerDialog.tsx:77]: Fixed in 4911942 — added aria-label="Remove" to preserve the accessible name during spinner display.

File: client/src/components/topology/ServerManagementSection.tsx [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+                        }}
+                    >
+                        {addingServer ? (
+                            <CircularProgress
+                                size={18}
+                                sx={{ color: 'inherit' }}
+                            />
+                        ) : (
+                            'Add'
+                        )}
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/topology/ServerManagementSection.tsx:156]: Fixed in 4911942 — added aria-label="Add server" to preserve the accessible name during spinner display.

File: client/src/components/__tests__/TopologyPanel.test.tsx [hunk 7]
```diff
@@ -0,0 +1,1700 @@
...
+                replicationType="binary"
+            />,
+        );
+
+        await waitFor(() => {
+            // Server names appear in both server list and relationship list
+            expect(screen.getAllByText('db-primary').length).toBeGreaterThanOrEqual(1);
+            expect(screen.getAllByText('db-standby').length).toBeGreaterThanOrEqual(1);
+            // "streams from" appears in relationship list and in Type dropdown
+            expect(screen.getAllByText(/streams from/i).length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/__tests__/TopologyPanel.test.tsx]: Fixed in 4911942 — the test now asserts via the relationship-specific delete button aria-label (Remove relationship between db-primary and db-standby), which only appears in the relationship list.

File: client/src/components/AdminPanel/tokens/__tests__/CreatedTokenDialog.test.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    it('does not render when open is false', () => {
+        renderComponent({ open: false });
+        expect(screen.queryByText('Token created')).not.toBeInTheDocument();
+    });
+
+    it('displays null token gracefully', () => {
+        renderComponent({ token: null });
+        // The dialog should still render, just without the token text
+        expect(screen.getByText('Token created')).toBeInTheDocument();
+    });
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/AdminPanel/tokens/__tests__/CreatedTokenDialog.test.tsx:107]: Fixed in 4911942 — added expect(screen.queryByText(TEST_TOKEN)).not.toBeInTheDocument() negative assertion.

File: client/src/components/AdminPanel/tokens/__tests__/CreateTokenDialog.test.tsx [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+        fireEvent.focus(addConnInput);
+        fireEvent.keyDown(addConnInput, { key: 'ArrowDown' });
+
+        await waitFor(() => {
+            expect(screen.getByRole('option', { name: 'Primary DB' })).toBeInTheDocument();
+        });
+
+        fireEvent.click(screen.getByRole('option', { name: 'Primary DB' }));
+        expect(onScopedConnectionsChange).toHaveBeenCalled();
+    });
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/AdminPanel/tokens/__tests__/CreateTokenDialog.test.tsx:253]: Fixed in 4911942 — now asserts the full payload shape with expect.objectContaining({ id, name, access_level }).

File: client/src/components/AdminPanel/tokens/__tests__/ScopeMultiSelect.test.tsx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+        // Find the chip's delete button (the X on the "All Options" chip)
+        const chip = screen.getByText('All Options').closest('.MuiChip-root');
+        expect(chip).toBeInTheDocument();
+
+        const deleteButton = chip?.querySelector('[data-testid="CancelIcon"]');
+        if (deleteButton) {
+            fireEvent.click(deleteButton);
+            expect(onChange).toHaveBeenCalled();
+        }
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/AdminPanel/tokens/__tests__/ScopeMultiSelect.test.tsx]: Fixed in 4911942 — replaced the if-guard with expect(deleteButton).toBeTruthy() so the test fails fast if the selector breaks.

File: client/src/components/__tests__/TopologyPanel.test.tsx [hunk 11]
```diff
@@ -0,0 +1,1620 @@
...
+                clusterName="test-cluster"
+                replicationType="binary"
+            />,
+        );
+
+        await waitFor(() => {
+            expect(
+                screen.getByText('Failed to load cluster topology'),
+            ).toBeInTheDocument();
+        });
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/__tests__/TopologyPanel.test.tsx:179]: Fixed in dd8ce45. Both error tests now use `mockApiGet.mockImplementation` to selectively reject only topology requests while letting `/api/v1/connections` succeed.
